modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8564,7 +8563,9 @@ final void setTreeVisible(boolean value) {
             if (treeVisible && !isDirtyEmpty()) {
                 addToSceneDirtyList();
             }
-            ((TreeVisiblePropertyReadOnly) treeVisibleProperty()).invalidate();
+            if (treeVisibleProperty != null) {
```

Comment:
Yeah, I think this is what was originally intended here; it's pointless to "create" the property just to invalidate it, as there is no one listening, because if there was someone listening, the property would have to have been created already.  So this is causing the property to be created by default for all nodes without a good reason.